lang/src/loader_account.rs

```diff
@@ -206,3 +160,51 @@ impl<'info, T: ZeroCopy + Owner> Key for AccountLoader<'info, T> {
         *self.acc_info.key
     }
 }
+
+///
+/// # Safety
+///
+/// Dereferencing the self.data raw pointer is
+/// safe because it will always point to valid data.
+/// This is because after the instantiation
+/// of an AccountLoader the location of the data the pointer
+/// is pointing to
+/// (the memory location of the &[u8] inside the inner AccountInfo)
+/// will not change and the data will not be freed
+/// before the pointer is freed; the pointer drops when
+/// AccountLoader is dropped but the &[u8] will always
+/// live until the end of the transaction
+///
+impl<'info, T> Deref for AccountLoader<'info, T>
+where
+    T: ZeroCopy + Owner,
+{
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+        unsafe { &(*self.data) }
+    }
+}
+
+///
+/// # Safety
+///
+/// Dereferencing the self.data raw pointer is
+/// safe because it will always point to valid data.
+/// This is because after the instantiation
+/// of an AccountLoader the location of the data the pointer
+/// is pointing to
+/// (the memory location of the &[u8] inside the inner AccountInfo)
+/// will not change and the data will not be freed
+/// before the pointer is freed; the pointer drops when
+/// AccountLoader is dropped but the &[u8] will always
+/// live until the end of the transaction
+///
+impl<'info, T> DerefMut for AccountLoader<'info, T>
+where
+    T: ZeroCopy + Owner,
+{
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        unsafe { &mut (*self.data) }
+    }
```

Comment:
It would be nice to have `anchor-debug` feature here. https://github.com/project-serum/anchor/blob/e8934a113c48db991518ec3b566fa5027273a3bb/lang/src/account.rs#L169

Reply:
We should probably just remove the feature flag and always do the check, even though I originally asked for the feature flag :).

Reply:
I could also just leave it in like it is now for consistency with the other account types because the check in derefmut will be removed anyway once the Mut types arrive (they'll have it in the constructor). but depends on when that feature gets released